client/state/sites/vouchers/reducer.js

```diff
@@ -28,6 +33,28 @@ import { itemsSchema } from './schema';
 export const items = ( state = {}, action ) => {
 	const { siteId } = action;
 	switch ( action.type ) {
+		case SITE_VOUCHERS_ASSIGN:
+			switch ( action.serviceType ) {
+				case GOOGLE_AD_CREDITS:
+					const currentVouchers = state[ siteId ]
+						? ( state[ siteId ][ GOOGLE_AD_CREDITS ] || [] )
+						: [];
+
+					const vouchers = currentVouchers.splice( 0 );
+					vouchers.push( action.voucher );
+
+					return Object.assign(
+						{},
+						state,
+						{
+							[ siteId ]: Object.assign(
+								{},
+								{ [ GOOGLE_AD_CREDITS ]: vouchers }
+							)
```

Comment:
a quick note here: I'm not sure why we need to be cloning over and over here. We cloned the input coming in, so it's normal to return the object itself. This should be good enough:

```js
return Object.assign( {}, state, {
	[ siteId ]: { [ GOOGLE_AD_CREDITS ]: vouchers }
} );
```